### PR TITLE
Push Notification Changes + Project Fixes + Dark Mode Support

### PR DESCRIPTION
**Pull request by droobah**
- Adds support for Dark Mode in iOS 13+
- Fixes absolute paths in project
- Includes changes for push notifications for the following
  1. Push notifications for all app states (foreground, background, terminated)
  2. Push notification actions (requires change to FirebaseProvider.py to set apns->payload->aps->category = PUSH_AUTHENTICATION in the message)
  3. Push notification actions work from watchOS as well

### CODE REVIEW

File: privacyIDEA Authenticator.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -825,16 +809,16 @@
...
-				DEVELOPMENT_TEAM = 627QALYL3B;
+				DEVELOPMENT_TEAM = 52NSLTN3S4;
 				INFOPLIST_FILE = "privacyIDEA Authenticator/Info.plist";
 				IPHONEOS_DEPLOYMENT_TARGET = 12.4;
 				LD_RUNPATH_SEARCH_PATHS = (
 					"$(inherited)",
 					"@executable_path/Frameworks",
 				);
-				PRODUCT_BUNDLE_IDENTIFIER = privacyidea.authenticator;
+				PRODUCT_BUNDLE_IDENTIFIER = net.cowism.privacyidea.authenticator;
```
Comment (nilsbehlen) [inline privacyIDEA Authenticator.xcodeproj/project.pbxproj]: we probably don't want a changed identifier

**Review by nilsbehlen — COMMENTED** [review]
can u add a PR to privacyidea aswell for changing the message payload?

**Comment by nilsbehlen** [comment]
I tried it out and it looks good to me, but i didn't get notifications in terminated state (ios 12.4). Are the changes to the firebaseprovider.py needed for that?

File: privacyIDEA Authenticator.xcodeproj/project.pbxproj (hunk 1 above)
Comment (droobah) [reply privacyIDEA Authenticator.xcodeproj/project.pbxproj]: Yeah, sorry, I had to change that to get it to compile and forgot to revert/cherry-pick the change.

**Comment by droobah** [comment]
> I tried it out and it looks good to me, but i didn't get notifications in terminated state (ios 12.4). Are the changes to the firebaseprovider.py needed for that?

Yes, the firebaseprovider.py changes are needed for notifications to work in a terminated state.  The changes switch from silent push to alert notifications with all of the required/recommended APNS flags.  I need to do a couple tests to ensure it doesn't affect the Android push notifications.  Once I confirm it won't affect the Android app, I'll toss up a PR for that.  In the meantime, I put the payload change in this comment so you can test yourself.

Also, there appears to be a timing bug that pops up where the loadPIPublicKey function will fail between 10-25% of the time when the app is in the background/terminated during a response to a push notification action.  I'm currently working on that and hopefully will update this PR today with the fix.

```
        fcm_message = {
            "message": {
                        "data": data,
                        "token": firebase_token,
                        "notification": {
                                         "title": data.get("title"),
                                         "body": data.get("question"),
                                        },
                         "android": {
                                     "priority": "HIGH",
                                     "ttl": "120s",
                                     "notification": {
                                                      "sound": "default",
                                                      "tag": "privacyidea.pushtoken",
                                                      "notification_priority": "PRIORITY_MAX"
                                                     },
                                     "fcm_options": { "analytics_label": "AndroidPushToken" }
                                    },
                         "apns": {
                                  "headers": {
                                              "apns-priority": "10",
                                              "apns-push-type": "alert",
                                              "apns-collapse-id": "privacyidea.pushtoken",
                                              "apns-expiration": str(int(time.time())+120)
                                             },
                                  "payload": {
                                              "aps": {
                                                      "sound": "default",
                                                      "category": "PUSH_AUTHENTICATION"
                                                     }
                                             },
                                  "fcm_options": { "analytics_label": "iOSPushToken" }
                                 }
                       }
            }

```

**Comment by droobah** [comment]
Okay, the background bug is fixed.  Had to change the KeychainSwift accessibility to allow background access.  Any existing push tokens will need to be re-enrolled to fix the issue.

**Comment by droobah** [comment]
Okay, I believe I'm done making changes.  Everything appears to be working swimmingly.

**Comment by nilsbehlen** [comment]
thanks alot for your help!

**Pull request merged**